File: test_files/enum_test.c
// Simple enum test file
#include <stdio.h>

// Basic enum
enum Color {
    RED,
    GREEN,
    BLUE
};

// Enum with explicit values
enum Status {
    ERROR = -1,
    SUCCESS = 0,
    PENDING = 1
};

// Enum with mixed values
enum Priority {
    LOW = 1,
    MEDIUM,     // Should be 2
    HIGH = 10,
    CRITICAL    // Should be 11
};

// Function using enums
void print_color(enum Color c) {
    switch(c) {
        case RED:
            printf("Red\n");
            break;
        case GREEN:
            printf("Green\n");
            break;
        case BLUE:
            printf("Blue\n");
            break;
    }
}

int main() {
    enum Color favorite = RED;
    enum Status app_status = SUCCESS;
    
    print_color(favorite);
    printf("Status: %d\n", app_status);
    
    return 0;
}

File: test_files/enum_test.rs
// Simple enum test file equivalent in Rust

// Basic enum
#[derive(Debug, Clone, Copy, PartialEq)]
pub enum Color {
    Red,
    Green,
    Blue,
}

// Enum with explicit values
#[derive(Debug, Clone, Copy, PartialEq)]
#[repr(i32)]
pub enum Status {
    Error = -1,
    Success = 0,
    Pending = 1,
}

// Enum with mixed values
#[derive(Debug, Clone, Copy, PartialEq)]
#[repr(i32)]
pub enum Priority {
    Low = 1,
    Medium = 2,
    High = 10,
    Critical = 11,
}

// Function using enums
pub fn print_color(c: Color) {
    match c {
        Color::Red => println!("Red"),
        Color::Green => println!("Green"),
        Color::Blue => println!("Blue"),
    }
}

fn main() {
    let favorite = Color::Red;
    let app_status = Status::Success;

    print_color(favorite);
    println!("Status: {}", app_status as i32);
}
